### PR TITLE
[Feature-1641][admin,client] Add Flink row level permissions

### LINKED ISSUE #1641: [Feature][admin,client] Add Flink row level permissions
### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

_No response_

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by aiwenmo**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dlink-admin/src/main/java/com/dlink/controller/RoleSelectPermissionsController.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RestController
+@RequestMapping("/api/roleSelectPermissions")
+public class RoleSelectPermissionsController {
+
+    @Autowired
+    private RoleSelectPermissionsService roleSelectPermissionsService;
```
Comment (leechor) [inline dlink-admin/src/main/java/com/dlink/controller/RoleSelectPermissionsController.java:48]: constructor inject

File: dlink-admin/src/main/java/com/dlink/controller/RoleSelectPermissionsController.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * create or update roleSelectPermissions
+     */
+    @PutMapping
+    public Result saveOrUpdateRole(@RequestBody RoleSelectPermissions roleSelectPermissions) {
+        if (roleSelectPermissionsService.saveOrUpdate(roleSelectPermissions)) {
+            return Result.succeed("新增成功");
+        } else {
+            return Result.failed("新增失败");
```
Comment (leechor) [inline dlink-admin/src/main/java/com/dlink/controller/RoleSelectPermissionsController.java:58]: english?

File: dlink-admin/src/main/java/com/dlink/controller/RoleSelectPermissionsController.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        }
+    }
+
+    /**
+     * delete roleSelectPermissions by id
+     */
+    @DeleteMapping
+    public Result deleteMul(@RequestBody JsonNode para) {
```
Comment (leechor) [inline dlink-admin/src/main/java/com/dlink/controller/RoleSelectPermissionsController.java:66]: rename

File: dlink-admin/src/main/java/com/dlink/service/impl/StudioServiceImpl.java [hunk 4]
```diff
@@ -152,6 +155,19 @@ private void addFlinkSQLEnv(AbstractStatementDTO statementDTO) {
                 process.info("No FlinkSQLEnv are loaded.");
             }
         }
+
+        process.info("Initializing data permissions...");
+        List<RoleSelectPermissions> currentRoleSelectPermissions = userService.getCurrentRoleSelectPermissions();
+        if (Asserts.isNotNullCollection(currentRoleSelectPermissions)) {
```
Comment (leechor) [inline dlink-admin/src/main/java/com/dlink/service/impl/StudioServiceImpl.java:161]: here  need lock to keep thread safe.

File: dlink-admin/src/main/java/com/dlink/service/impl/TaskServiceImpl.java [hunk 5]
```diff
@@ -963,6 +969,17 @@ else if (Dialect.KUBERNETES_APPLICATION.equalsVal(task.getDialect())
                 config.setSavePointPath(null);
         }
         config.setVariables(fragmentVariableService.listEnabledVariables());
+        List<RoleSelectPermissions> currentRoleSelectPermissions = userService.getCurrentRoleSelectPermissions();
```
Comment (leechor) [inline dlink-admin/src/main/java/com/dlink/service/impl/TaskServiceImpl.java:972]: here is duplication, seemed should extract to a single class.

File: dlink-admin/src/main/java/com/dlink/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -77,6 +80,9 @@ public class UserServiceImpl extends SuperServiceImpl<UserMapper, User> implemen
     @Autowired
     private TenantService tenantService;
 
+    @Autowired
+    private RoleSelectPermissionsService roleSelectPermissionsService;
```
Comment (leechor) [inline dlink-admin/src/main/java/com/dlink/service/impl/UserServiceImpl.java:84]: contructor inject

File: dlink-admin/src/main/resources/mapper/RoleSelectPermissionsMapper.xml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    <select id="selectForProTable" resultType="com.dlink.model.RoleSelectPermissions">
+        select
+        a.*,
+        b.role_code,
+        b.role_name
+        from
+        dlink_role_select_permissions a
+        left join dlink_role b on b.id = a.role_id
+        <where>
+            1=1
```
Comment (leechor) [inline dlink-admin/src/main/resources/mapper/RoleSelectPermissionsMapper.xml:18]: 1=1 needn't already.<where>

**Comment by aiwenmo** [comment]
This is 0.7 branch.
Code optimization is performed in dev branch.
This function will not be added until the dev branch refactoring is completed.

**Pull request merged**